### PR TITLE
ENH: Add ``copy_header`` inputs to some ANTs interfaces

### PR DESCRIPTION
**Pull request by oesteban**
This PR adds an input ``copy_header`` to the ``ImageMath`` and ``ThresholdImage`` interfaces because ANTs fails to produce images with appropriate x-form headers.

In particular, these changes are necessary for poldracklab/mriqc#803 to go forward.

In addition to the ``copy_header`` change, some PEP stylistic issues have been addressed.

### CODE REVIEW

**Comment by effigies** [comment]
Are we testing this anywhere downstream? `super()._list_outputs()` is not guaranteed to work for all interfaces, so that we're not getting coverage on these lines worries me. A cheap test that exercises this functionality for each interface would be nice.

**Comment by effigies** [comment]
Otherwise, this looks good.

File: niworkflows/interfaces/ants.py [hunk 1]
```diff
@@ -516,3 +534,16 @@ def _list_outputs(self):
...
             )
         return outputs
+
+
+def _copy_header(header_file, in_file, set_dtype=True):
+    """Copy header from input image to an output image."""
+    import nibabel as nb
+    in_img = nb.load(header_file)
+    out_img = nb.load(in_file, mmap=False)
+    new_img = out_img.__class__(out_img.get_fdata(), in_img.affine,
```
Comment (oesteban) [inline niworkflows/interfaces/ants.py]: @effigies - shouldn't this be using the `out_img.dataobj` instead?

Comment (effigies) [reply niworkflows/interfaces/ants.py]: I would. It shouldn't change the values, just skip a cast or two.

**Pull request merged**